lib/attrsets.nix

```diff
@@ -518,6 +518,27 @@ rec {
     set:
     listToAttrs (map (attr: f attr set.${attr}) (attrNames set));
 
+  /* Like mapAttrs, but additionally taking a variable that
+     will be automatically incremented, starting from 0.
+
+     Example:
+       imapAttrs (i: name: value: value + i)
+          { a = 1; b = 1; c = 1; }
+       => { a = 1; b = 2; c = 3; }
+  */
+  imapAttrs = f: set:
+    listToAttrs (imap0 (i: attr: nameValuePair attr (f i attr set.${attr})) (attrNames set));
+
+  /* Like mapAttrs', but additionally taking a variable that
+     will be automatically incremented, starting from 0.
+
+     Example:
+       imapAttrs' (i: name: value: nameValuePair "${name}${toString i}" value + i)
```

Comment:
Both functions have the same signature? If so, what is the difference between them?

Reply:
they apply f in different places, see the examples:

`imapAttrs` has a built in nameValuePair, calling f on just the value, you can't change the names from the input attrset.
`imapAttrs'` wants a function that provides a nameValuePair, so you can modify both the name and the value of the input attrset.